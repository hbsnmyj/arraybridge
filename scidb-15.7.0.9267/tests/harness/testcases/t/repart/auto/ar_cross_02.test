# If the right argument is explicitly repartitioned so as not to match
# corresponding dimensions of the left argument, make sure we see an
# error.

--setup

create array left_array<val:double>[i=0:9,5,0, j=0:9,5,0]

create array right_array<val:double>[k=0:19,10,4]
create array right_ok   <val:double>[k=0:29,5,0]
create array right_bad  <val:double>[k=0:19,5,2]

--start-igdata
store(build(left_array,i*3+j),left_array)
store(build(right_array,k+101),right_array)
--stop-igdata

--test

--start-query-logging

# No explicit repart, all is fine.
--start-igdata
cross_join(left_array,right_array,j,k)
--stop-igdata

--error --code=scidb::SCIDB_SE_OPERATOR::SCIDB_LE_BAD_EXPLICIT_REPART "cross_join(left_array, repart(right_array, right_bad), j, k)"

# No problem if explict repart is what's wanted anyhow.
--start-igdata
cross_join(left_array, repart(right_array, right_ok), j, k)
--stop-igdata

--stop-query-logging

--cleanup

remove(left_array)
remove(right_array)
remove(right_ok)
remove(right_bad)
