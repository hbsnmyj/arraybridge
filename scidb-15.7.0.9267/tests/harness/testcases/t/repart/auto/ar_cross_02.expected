Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

[Query was executed successfully, ignoring data output by this query.]

[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <cross_join(left_array,right_array,j,k)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <cross_join(left_array, repart(right_array, right_bad), j, k)>
[An error expected at this place for the query "cross_join(left_array, repart(right_array, right_bad), j, k)". And it failed with error code = scidb::SCIDB_SE_OPERATOR::SCIDB_LE_BAD_EXPLICIT_REPART. Expected error code = scidb::SCIDB_SE_OPERATOR::SCIDB_LE_BAD_EXPLICIT_REPART.]

SCIDB QUERY : <cross_join(left_array, repart(right_array, right_ok), j, k)>
[Query was executed successfully, ignoring data output by this query.]

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

